pkg/controller/masteruserrecord/sync.go

```diff
@@ -205,14 +207,33 @@ func (s *Synchronizer) openShift3XConsoleURL(apiEndpoint string) (string, error)
 	return url, nil
 }
 
+func (s *Synchronizer) alignDisabled() {
+	for _, uaStatus := range s.record.Status.UserAccounts {
+		if !isDisabled(uaStatus.Conditions) {
+			return
+		}
+	}
+	s.record.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(s.record.Status.Conditions, toBeDisabled())
+}
+
+func isDisabled(conditions []toolchainv1alpha1.Condition) bool {
+	for _, con := range conditions {
+		if con.Type == toolchainv1alpha1.ConditionReady {
+			return con.Reason == toolchainv1alpha1.UserAccountDisabledReason
+		}
+	}
+	return false
+}
+
 // alignReadiness checks if all embedded SAs are ready
-func (s *Synchronizer) alignReadiness() {
+func (s *Synchronizer) alignReadiness() bool {
 	for _, uaStatus := range s.record.Status.UserAccounts {
 		if !isReady(uaStatus.Conditions) {
-			return
+			return false
 		}
 	}
 	s.record.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(s.record.Status.Conditions, toBeProvisioned())
+	return true
 }
 
 func isReady(conditions []toolchainv1alpha1.Condition) bool {
```

Comment:
I know you are not changing this code but since you are editing the file anyway.. let's replace this `isReady()` function by https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/condition/condition.go#L39